safekeeper/src/send_wal.rs

```diff
@@ -601,15 +602,20 @@ impl<IO: AsyncRead + AsyncWrite + Unpin> WalSender<'_, IO> {
             };
             let send_buf = &send_buf[..send_size];
 
-            // and send it
-            self.pgb
-                .write_message(&BeMessage::XLogData(XLogDataBody {
-                    wal_start: self.start_pos.0,
-                    wal_end: self.end_pos.0,
-                    timestamp: get_current_timestamp(),
-                    data: send_buf,
-                }))
-                .await?;
+            // and send it, while respecting Timeline::cancel
+            let msg = BeMessage::XLogData(XLogDataBody {
+                wal_start: self.start_pos.0,
+                wal_end: self.end_pos.0,
+                timestamp: get_current_timestamp(),
+                data: send_buf,
+            });
+            tokio::select! {
```

Comment:
Same as with the WAL receiver, I think we can move the token select up into `handle_start_replication_guts` and drop the future. That goes for this whole file.

Reply:
Done in c8623b55ff6b5ff051a5f8f2d3c08e8a12015a1b